### PR TITLE
Rename rzpm to rz-pm for consistency with rz-asm, rz-bin, etc.

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

**Comment by XVilka** [comment]
```
Error:     rizin_test.go:39: could not checkout "1.0.0": exit status 1
=== RUN   TestRizin/UninstallRizin
--- FAIL: TestRizin (7.76s)
    --- FAIL: TestRizin/InstallRizin (7.74s)
    --- PASS: TestRizin/UninstallRizin (0.00s)
```

**Comment by ret2libc** [comment]
This is not a rzpm problem. It's just that we don't have a release yet and i don't know what number to use there..

**Review by ITAYC0HEN — APPROVED** [review]
Looks good to me

**Comment by ret2libc** [comment]
Wait for this please, i need to check w bunch of stuff..

**Review by XVilka — APPROVED** [review]

**Comment by ret2libc** [comment]
I have changed the paths on Linux, however we need to test Windows and Unix. Also, I'd propose to split "Unix" into MacOS and generic "Unix" (e.g. *BSD, etc.), because probably MacOS has specific directories where to put applications.

I'm still not sure about the WIndows/Unix dirs, I think we can do something better there.

Also, as I did things right now, rz-pm will install Rizin in `~/.local`. That means that if a user already had Rizin installed in `~/.local` it will be overwritten. Does it make sense? Or shall we do anything else? The problem with installing somewhere else might be that the user will have to manually add paths to `PATH` variable. WDYT?

**Comment by ret2libc** [comment]
Maybe we should just put everything under a rz-pm specific directory (even rizin) and inform the user he has to change his PATH vars to run it. Otherwise we may collide with the user choices... WDYT?

For example, Rizin could be installed under `~/.local/share/rz-pm/install`. This means that `~/.local/share/rz-pm/install` is going to be the prefix used to build/install Rizin and everything related to rizin (e.g. plugins) will be under that dir. So, `rizin` binary would be in `~/.local/share/rz-pm/install/bin/rizin` and plugins would be in `~/.local/share/rz-pm/install/share/rizin/plugins`.

File: pkg/site/rizin_nix.go [hunk 1]
```diff
@@ -49,53 +46,30 @@ func (s Site) InstallRizin(prefix, version string) error {
...
-
-	if out, err := cmdMake.Output(); err != nil {
-		log.Print(string(out))
+	if res, err := process.Run("ninja", []string{"-C", buildDir}, srcDir); err != nil {
+		log.Print(string(res.Stdout.String()))
 		return err
 	}
 
-	if _, err := process.Run(makeBin, []string{"install"}, srcDir); err != nil {
+	if _, err := process.Run("ninja", []string{"-C", buildDir, "install"}, srcDir); err != nil {
```
Comment (XVilka) [inline pkg/site/rizin_nix.go:72]: If you switch to `ninja`, please fix the `jsdec` package to use it too: https://github.com/rizinorg/rz-pm-db/blob/master/db/jsdec#L7

I also suggest to add CMake-based rz-ghidra https://github.com/rizinorg/rizin-pm/blob/master/db/rz-ghidra in that database for testing.

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
> Rizin could be installed under ~/.local/share/rz-pm/install. This means that ~/.local/share/rz-pm/install is going to be the prefix used to build/install Rizin and everything related to rizin (e.g. plugins) will be under that dir. So, rizin binary would be in ~/.local/share/rz-pm/install/bin/rizin and plugins would be in ~/.local/share/rz-pm/install/share/rizin/plugins.

Yes, that would be better. Like `rustup` installs everything in `~/.rustup/`:

**Comment by XVilka** [comment]
Please remove windows downloading from http://radare.mikelloc.com:
```
2021/01/15 09:54:55 Downloading http://radare.mikelloc.com/get/dev/rizin-msvc_64-dev.zip into C:\Users\RUNNER~1\AppData\Local\Temp\rz-pm_856052973.zip 
Error:     rizin_test.go:39: could not create a zip reader: zip: not a valid zip file
```

File: pkg/site/rizin_nix.go (hunk 1 above)
Comment (wargio) [reply pkg/site/rizin_nix.go:72]: actually jsdec has also meson

**Comment by XVilka** [comment]
Hmm:
```
Error:     rizin_test.go:51: Could not stat("C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\253926610\\bin\\rizin.bat")
=== RUN   TestRizin/UninstallRizin
--- FAIL: TestRizin (5.35s)
    --- FAIL: TestRizin/InstallRizin (4.51s)
    --- PASS: TestRizin/UninstallRizin (0.00s)
=== RUN   TestNew
=== RUN   TestNew/C:\Users\RUNNER~1\AppData\Local\Temp\590375892\rz-pm\rz-pm-db
=== RUN   TestNew/C:\Users\RUNNER~1\AppData\Local\Temp\590375892\rz-pm\git
=== RUN   TestNew/C:\Users\RUNNER~1\AppData\Local\Temp\590375892\rz-pm\installed
--- PASS: TestNew (0.04s)
    --- PASS: TestNew/C:\Users\RUNNER~1\AppData\Local\Temp\590375892\rz-pm\rz-pm-db (0.00s)
    --- PASS: TestNew/C:\Users\RUNNER~1\AppData\Local\Temp\590375892\rz-pm\git (0.00s)
    --- PASS: TestNew/C:\Users\RUNNER~1\AppData\Local\Temp\590375892\rz-pm\installed (0.00s)
FAIL
FAIL	github.com/rizinorg/rz-pm/pkg/site	5.422s
FAIL
Error: mingw32-make: *** [Makefile:14: integration-tests] Error 1
Error: Process completed with exit code 1.
```

**Comment by ret2libc** [comment]
This needs a lot of love ç__ç

I know, I have to find some time for this before 0.2.0 and I will.

**Comment by fangxlmr** [comment]
I've go to say it fails to do integration test on macOS either

<details>
 <summary>make integration-test</summary>

```
$ uname -a
Darwin LEONXLFANG-MB0 19.6.0 Darwin Kernel Version 19.6.0: Tue Jan 12 22:13:05 PST 2021; root:xnu-6153.141.16~1/RELEASE_X86_64 x86_64
$ make integration-tests
go test -v -tags=integration ./...
?       github.com/rizinorg/rz-pm       [no test files]
?       github.com/rizinorg/rz-pm/internal/features     [no test files]
=== RUN   TestSiteDir
--- PASS: TestSiteDir (0.00s)
PASS
ok      github.com/rizinorg/rz-pm/internal/util/dir     (cached)
?       github.com/rizinorg/rz-pm/lib   [no test files]
?       github.com/rizinorg/rz-pm/pkg   [no test files]
?       github.com/rizinorg/rz-pm/pkg/database  [no test files]
=== RUN   TestInit
=== RUN   TestInit/create
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/315582295/.git rev-parse" in .
2021/03/29 11:01:33 Running "git init" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/315582295
=== RUN   TestInit/recreate,_force=false
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/866886346/.git rev-parse" in .
2021/03/29 11:01:33 Running "git init" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/866886346
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/866886346/.git rev-parse" in .
=== RUN   TestInit/recreate,_force=true
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/616346529/.git rev-parse" in .
2021/03/29 11:01:33 Running "git init" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/616346529
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/616346529/.git rev-parse" in .
2021/03/29 11:01:33 Running "git init" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/616346529
--- PASS: TestInit (0.08s)
    --- PASS: TestInit/create (0.02s)
    --- PASS: TestInit/recreate,_force=false (0.03s)
    --- PASS: TestInit/recreate,_force=true (0.04s)
=== RUN   TestOpen
=== RUN   TestOpen//dev/null
2021/03/29 11:01:33 Running "git --git-dir /dev/null/.git rev-parse" in .
=== RUN   TestOpen/not_a_git_repo
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/561961612/.git rev-parse" in .
=== RUN   TestOpen/should_work
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/447257211/.git rev-parse" in .
2021/03/29 11:01:33 Running "git init" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/447257211
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/447257211/.git rev-parse" in .
--- PASS: TestOpen (0.04s)
    --- PASS: TestOpen//dev/null (0.01s)
    --- PASS: TestOpen/not_a_git_repo (0.01s)
    --- PASS: TestOpen/should_work (0.03s)
=== RUN   TestRepository_AddRemote
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/666791838/.git rev-parse" in .
2021/03/29 11:01:33 Running "git init" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/666791838
2021/03/29 11:01:33 Running "git remote add random-name git@localhost.localdomain:test" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/666791838
2021/03/29 11:01:33 Running "git remote -v" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/666791838
--- PASS: TestRepository_AddRemote (0.03s)
=== RUN   TestRun
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/342855269/.git rev-parse" in .
2021/03/29 11:01:33 Running "git init" in /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/342855269
2021/03/29 11:01:33 Running "git --git-dir /var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/342855269/.git rev-parse" in .
--- PASS: TestRun (0.03s)
PASS
ok      github.com/rizinorg/rz-pm/pkg/git       (cached)
?       github.com/rizinorg/rz-pm/pkg/process   [no test files]
=== RUN   TestInstaller_Install
--- PASS: TestInstaller_Install (0.00s)
=== RUN   TestManifest_Verify
=== RUN   TestManifest_Verify/empty_name
=== RUN   TestManifest_Verify/empty_version
=== RUN   TestManifest_Verify/empty_description
=== RUN   TestManifest_Verify/empty_install_instructions
--- PASS: TestManifest_Verify (0.00s)
    --- PASS: TestManifest_Verify/empty_name (0.00s)
    --- PASS: TestManifest_Verify/empty_version (0.00s)
    --- PASS: TestManifest_Verify/empty_description (0.00s)
    --- PASS: TestManifest_Verify/empty_install_instructions (0.00s)
=== RUN   Test_getFetcher
=== RUN   Test_getFetcher/git
=== RUN   Test_getFetcher/zip
=== RUN   Test_getFetcher/not-a-fetcher
--- PASS: Test_getFetcher (0.00s)
    --- PASS: Test_getFetcher/git (0.00s)
    --- PASS: Test_getFetcher/zip (0.00s)
    --- PASS: Test_getFetcher/not-a-fetcher (0.00s)
=== RUN   TestFromYAML
--- PASS: TestFromYAML (0.00s)
=== RUN   Test_osFromGOOS
=== RUN   Test_osFromGOOS/darwin
=== RUN   Test_osFromGOOS/freebsd
=== RUN   Test_osFromGOOS/linux
=== RUN   Test_osFromGOOS/openbsd
=== RUN   Test_osFromGOOS/windows
--- PASS: Test_osFromGOOS (0.00s)
    --- PASS: Test_osFromGOOS/darwin (0.00s)
    --- PASS: Test_osFromGOOS/freebsd (0.00s)
    --- PASS: Test_osFromGOOS/linux (0.00s)
    --- PASS: Test_osFromGOOS/openbsd (0.00s)
    --- PASS: Test_osFromGOOS/windows (0.00s)
PASS
ok      github.com/rizinorg/rz-pm/pkg/rzpackage (cached)
?       github.com/rizinorg/rz-pm/pkg/rzpackage/fetchers        [no test files]
=== RUN   TestRizin
=== RUN   TestRizin/InstallRizin
    TestRizin/InstallRizin: rizin_test.go:35: The rizin version must be defined
=== RUN   TestRizin/UninstallRizin
--- FAIL: TestRizin (0.00s)
    --- FAIL: TestRizin/InstallRizin (0.00s)
    --- PASS: TestRizin/UninstallRizin (0.00s)
=== RUN   TestNew
=== RUN   TestNew//var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/203667457/rz-pm/rz-pm-db
=== RUN   TestNew//var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/203667457/rz-pm/git
=== RUN   TestNew//var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/203667457/rz-pm/installed
--- PASS: TestNew (0.00s)
    --- PASS: TestNew//var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/203667457/rz-pm/rz-pm-db (0.00s)
    --- PASS: TestNew//var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/203667457/rz-pm/git (0.00s)
    --- PASS: TestNew//var/folders/5t/p75g7bfn7f5gy9_y4pw25t6w0000gn/T/203667457/rz-pm/installed (0.00s)
FAIL
FAIL    github.com/rizinorg/rz-pm/pkg/site      0.185s
FAIL
make: *** [Makefile:14：integration-tests] Error 1
```

</details>

**Comment by ret2libc** [comment]
Yeah I still need to finish this :( sorry.

**Comment by fangxlmr** [comment]
@ret2libc What are we still missing here? I've got a few minutes and may help you on this. :)

**Comment by ret2libc** [comment]
> @ret2libc What are we still missing here? I've got a few minutes and may help you on this. :)

So, IIRC we need to choose once and for all the directory layout we intend to use on all major platforms. Right now I think it is a bit off, but from the top of my mind I remember we wanted something like `~/.local/share/rz-pm` on linux.. we said we should use something like what `rustup` does (which I'd need to check, because I do not know). I think once the directories layout is well thought we can move forward with everything else.

**Comment by fangxlmr** [comment]
And also, I notice that one could install rizin command globally with `sudo` permission which means all users could access it, instead `rz-pm` command merely install all plugins for current user cuasing a gap for other users.

I found this issue in base docker image. Since I'd like to have RzGhidra installed in base image as posted in rizinorg/rizin#1011, I tried to run `rz-pm init && rz-pm -i rz-ghidra` before `USER rizin` was set in Dockerfile, and this leads to user rizin have no global default rz-ghidra at all. So user rizin need to install rz-ghidra over again. This could happen in physical machine as well.

So I was wondering if we could have a global default directory to put all-user-accessable binaries and plugins. If one installed something locally in one's $HOME, then it will overwrite the behavior of global default. WDYT?

**Comment by ret2libc** [comment]
In bash-rz-pm there is a `-gi` flag to install plugins globally. I think we can have a flag in golang-rz-pm as well to install things system-wide (e.g. `--system` flag or something like that). It should be just a matter of changing some INSTALL_PREFIX or LIB_PREFIX or something like that when compiling the plugin through rz-pm.

**Comment by fangxlmr** [comment]
> In bash-rz-pm there is a `-gi` flag to install plugins globally. 

Ah, that's nice. I didn't notice and even use it before.

> I think we can have a flag in golang-rz-pm as well to install things system-wide (e.g. `--system` flag or something like that). It should be just a matter of changing some INSTALL_PREFIX or LIB_PREFIX or something like that when compiling the plugin through rz-pm.

Sounds good.

And one more concern, does Rizin could tell the difference between system-wide and user-wide plugins?
Say I've rz-ghidra pre-installed globally (e.g. `rz-pm -gi rz-ghidra`, and now I install rz-ghidra user-wide (e.g. `rz-pm -i rz-ghidra`). Then what's the behavior of Rizin? Does it call system-wide or user-wide rz-ghidra?

**Comment by ret2libc** [comment]
> And one more concern, does Rizin could tell the difference between system-wide and user-wide plugins?
> Say I've rz-ghidra pre-installed globally (e.g. `rz-pm -gi rz-ghidra`, and now I install rz-ghidra user-wide (e.g. `rz-pm -i rz-ghidra`). Then what's the behavior of Rizin? Does it call system-wide or user-wide rz-ghidra?

Home plugins are loaded first, so if there is another plugin with the same name in the system-wide directory, it won't be loaded. See https://github.com/rizinorg/rizin/blob/dev/librz/util/lib.c#L267

**Review by ret2libc — COMMENTED** [review]
Ok, I don't want to delay this anymore. It has waited for too long.. it is far from perfect and the tool is not working at all, so no need to be perfect at this point. I will develop a lot on this in the next week, but for the moment i guess the renaming rzpm -> rz-pm is what i want to be included.

There are many more things to consider, like the proper paths for plugins, usage of git libraries instead of git binary, etc....

**Pull request merged**